app/domain/authentication/authn_k8s/web_socket_client_event_handler.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnK8s
+
+    class WebSocketClientEventHandler
+
+      attr_reader :channel_closed, :message_log
+
+      def initialize(
+        ws_client:,
+        stdin:,
+        body:,
+        pod_name:,
+        validate_message:,
+        message_log_class:,
+        logger:
+      )
+        @ws_client = ws_client
+        @stdin = stdin
+        @body = body
+        @pod_name = pod_name
+        @validate_message = validate_message
+        @message_log_class = message_log_class
+        @logger = logger
+
+        @message_log = @message_log_class.new
+        @channel_closed = false
+      end
+
+      def on_open
+        hs       = @ws_client.handshake
+        hs_error = hs.error
+
+        if hs_error
+          # Add the error to the stream instead of raising it so we continue to
+          # receive other messages. At the end of this class run we will verify
+          # the stream to see if any errors were written
+          @ws_client.emit(
+            :error,
+            Errors::Authentication::AuthnK8s::WebSocketHandshakeError.new(
+              hs_error.inspect
+            )
+          )
+          return
+        end
+
+        @logger.debug(
+          LogMessages::Authentication::AuthnK8s::PodChannelOpen.new(@pod_name)
+        )
+
+        return unless @stdin
+
+        # stdin was provided. We send it to the client.
+
+        data = WebSocketMessage.channel_byte('stdin') + @body
+        @ws_client.send(data)
+
+        # We close the socket and don't wait for the other side to close it
+        # so that we can finish handling the request quickly and don't leave the
+        # Conjur server hanging. If an error occurred it will be written to
+        # the client container logs.
+        @ws_client.send(nil, type: :close)
```

Comment:
I'm confused by this comment.  It looks like this logic always runs, so we're sending data on line 57 and then immediately the server is hanging and we send a close?  I'm sure I'm missing something but some clarification would be helpful.

Reply:
yes, you described exactly what happens and it is explained in the comment. what are you missing? i can enhance the comment according to your misunderstanding.